remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/mapping/reference/MessageReferenceProvider.java

```diff
@@ -21,15 +24,19 @@ public MessageReferenceProvider(Descriptor descriptor) {
       this.fields = new ArrayList<>(descriptor.getFields().size());
       this.embedded = new ArrayList<>();
 
-      IndexingMetadata indexingMetadata = descriptor.getProcessedAnnotation(IndexingMetadata.INDEXED_ANNOTATION);
+      IndexingMetadata indexingMetadata = descriptor.getProcessedAnnotation(INDEXED_ANNOTATION);
       if (indexingMetadata == null) {
          return;
       }
 
       for (FieldDescriptor fieldDescriptor : descriptor.getFields()) {
          String fieldName = fieldDescriptor.getName();
          if (Type.MESSAGE.equals(fieldDescriptor.getType())) {
-            embedded.add(new Embedded(fieldName, fieldDescriptor.getMessageType().getFullName(), fieldDescriptor.isRepeated()));
+            if (fieldDescriptor.getAnnotations().containsKey(FIELD_ANNOTATION)) {
+               if (hasFieldAnnotation(fieldDescriptor.getMessageType())) {
+                  embedded.add(new Embedded(fieldName, fieldDescriptor.getMessageType().getFullName(), fieldDescriptor.isRepeated()));
```

Comment:
IIUC this is going to add an embedded type if the type of the field has any annotated field, regardless of whether it's stored/analyzed/indexed.

I'd like to see a comment in the code explaining why the annotation matters but its attributes don't, even if it's obvious to someone more familiar with Hibernate Search than myself.

Reply:
A @Field pseudo annotation signals that the member is managed by hibernate search. The attributes of the pseudo annotation is all managed by Hibernate Search, the job of this class is to filter out protobuf members that are not eligible to be managed by hibernate search.

Reply:
Sure, that makes sense.

The problem is that the class and method have no javadoc, and it doesn't implement any interface, so it's not obvious what its job is.

Reply:
Indeed, also to me it was not immediate clear what it was doing, I will add some javadoc

Reply:
fixed